File: java/ucotron-sdk/src/main/java/com/ucotron/sdk/types/CreateMemoryResult.java
package com.ucotron.sdk.types;

import java.util.List;

/**
 * Result of creating a new memory.
 */
public class CreateMemoryResult {
    private List<Long> chunk_node_ids;
    private List<Long> entity_node_ids;
    private int edges_created;
    private IngestionMetrics metrics;

    public List<Long> getChunkNodeIds() { return chunk_node_ids; }
    public List<Long> getEntityNodeIds() { return entity_node_ids; }
    public int getEdgesCreated() { return edges_created; }
    public IngestionMetrics getMetrics() { return metrics; }
}

File: php/src/Types/CreateMemoryResult.php
<?php

declare(strict_types=1);

namespace Ucotron\Sdk\Types;

class CreateMemoryResult
{
    public function __construct(
        public readonly string $id,
        public readonly ?IngestionMetrics $metrics = null,
    ) {}

    public static function fromArray(array $data): self
    {
        return new self(
            id: (string) ($data['id'] ?? ''),
            metrics: isset($data['metrics']) ? IngestionMetrics::fromArray($data['metrics']) : null,
        );
    }
}
